test/catalyst/assetAttributesRegistry/assetAttributesRegistry.test.ts

```diff
@@ -39,16 +40,30 @@ describe('AssetAttributesRegistry', function () {
   });
 
   it('setCatalyst for legendary catalyst with 4 gems', async function () {
-    const assetId = BigNumber.from('0x1fe80000800000000000000000000000');
+    const {
+      assetUpgrader,
+      assetAttributesRegistry,
+    } = await setupAssetAttributesRegistry();
     const users = await getUnnamedAccounts();
+    const assetId = await mintAsset(
+      users[0],
+      BigNumber.from('2233'),
+      '0x1111111111111111111111111111111111111111111111111111111111111111',
+      1,
+      0,
+      users[0],
+      Buffer.from('ff')
+    );
     const legendaryCatalystId = catalysts[3].catalystId;
     const gemsIds = gems.filter((gem) => gem.gemId < 5).map((gem) => gem.gemId);
     const {record, event, block} = await setCatalyst(
```

Comment:
@wighawag As you can see, `setCatalyst` function is now using AssetUpgrader, which is fine, but it means it's no longer a unit-test. I'm in favor of leaving it as is, as long as the solidity coverage works well for cross-contract transactions, which will have to be determined.

Reply:
I am fine with using integration test. the setCatalyst is not meant to be used by user themselves. it will always have to go through either AssetUpgrader or AssetUpdated.

We could always re add the previous test though. Did the previous test added a EOA as AssetUpgrader ?

Reply:
yes, it was an EOA